blockmanager.go

```diff
@@ -186,23 +190,36 @@ type blockManager struct {
 	started           int32
 	shutdown          int32
 	blockChain        *blockchain.BlockChain
-	requestedTxns     map[wire.ShaHash]struct{}
-	requestedBlocks   map[wire.ShaHash]struct{}
 	progressLogger    *blockProgressLogger
 	receivedLogBlocks int64
 	receivedLogTx     int64
 	processingReqs    bool
-	syncPeer          *peer
+	syncPeer          *peer.Peer
 	msgChan           chan interface{}
 	chainState        chainState
 	wg                sync.WaitGroup
 	quit              chan struct{}
+	txProcessed       chan struct{}
+	blockProcessed    chan struct{}
+	peerProcessed     chan struct{}
 
 	// The following fields are used for headers-first mode.
 	headersFirstMode bool
 	headerList       *list.List
 	startHeader      *list.Element
 	nextCheckpoint   *chaincfg.Checkpoint
+	peers            map[int32]*peerData
+}
+
+// peerData contains the data associated with each peer
+type peerData struct {
+	requestQueue    []*wire.InvVect
+	requestedTxns   map[wire.ShaHash]struct{}
+	requestedBlocks map[wire.ShaHash]struct{}
+
+	continueHash   *wire.ShaHash
+	disableRelayTx bool
+	relayMtx       sync.Mutex
```

Comment:
Super minor nitpick, but typically a mutex is defined before the thing(s) it protects.